File: supabase_client.py
from supabase import create_client, Client
import os
from typing import Optional, List, Dict, Any

# Supabase configuration
SUPABASE_URL = "https://dmskwpquomqcsumaqaej.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImRtc2t3cHF1b21xY3N1bWFxYWVqIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Mzg5Mzk0NDMsImV4cCI6MjA1NDUxNTQ0M30.A6aMcEqnl2V7l4vlwG9dUtK7F4BBIT3K4Lz7L-rHTd4"

# Initialize Supabase client
supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)

class SupabaseService:
    """Service class for Supabase operations"""
    
    def __init__(self):
        self.client = supabase
    
    # User operations
    async def create_user(self, user_data: Dict[str, Any]) -> Optional[Dict[str, Any]]:
        """Create a new user in Supabase"""
        try:
            result = self.client.table('users').insert(user_data).execute()
            return result.data[0] if result.data else None
        except Exception as e:
            print(f"Error creating user: {e}")
            return None
    
    async def get_user_by_firebase_uid(self, firebase_uid: str) -> Optional[Dict[str, Any]]:
        """Get user by Firebase UID"""
        try:
            result = self.client.table('users').select('*').eq('firebase_uid', firebase_uid).execute()
            return result.data[0] if result.data else None
        except Exception as e:
            print(f"Error getting user: {e}")
            return None
    
    async def update_user(self, user_id: str, updates: Dict[str, Any]) -> Optional[Dict[str, Any]]:
        """Update user data"""
        try:
            result = self.client.table('users').update(updates).eq('id', user_id).execute()
            return result.data[0] if result.data else None
        except Exception as e:
            print(f"Error updating user: {e}")
            return None
    
    # Restaurant operations
    async def get_all_restaurants(self) -> List[Dict[str, Any]]:
        """Get all restaurants"""
        try:
            result = self.client.table('restaurants').select('*').order('name').execute()
            return result.data or []
        except Exception as e:
            print(f"Error getting restaurants: {e}")
            return []
    
    async def get_restaurant_by_id(self, restaurant_id: str) -> Optional[Dict[str, Any]]:
        """Get restaurant by ID"""
        try:
            result = self.client.table('restaurants').select('*').eq('id', restaurant_id).execute()
            return result.data[0] if result.data else None
        except Exception as e:
            print(f"Error getting restaurant: {e}")
            return None
    
    async def search_restaurants(self, query: str) -> List[Dict[str, Any]]:
        """Search restaurants by name"""
        try:
            result = self.client.table('restaurants').select('*').ilike('name', f'%{query}%').execute()
            return result.data or []
        except Exception as e:
            print(f"Error searching restaurants: {e}")
            return []
    
    async def get_restaurants_by_category(self, category: str) -> List[Dict[str, Any]]:
        """Get restaurants by category"""
        try:
            result = self.client.table('restaurants').select('*').eq('category', category).execute()
            return result.data or []
        except Exception as e:
            print(f"Error getting restaurants by category: {e}")
            return []
    
    # Food operations
    async def get_foods_by_restaurant(self, restaurant_id: str) -> List[Dict[str, Any]]:
        """Get foods by restaurant ID"""
        try:
            result = self.client.table('foods').select('*').eq('restaurant_id', restaurant_id).execute()
            return result.data or []
        except Exception as e:
            print(f"Error getting foods: {e}")
            return []
    
    async def get_food_by_id(self, food_id: str) -> Optional[Dict[str, Any]]:
        """Get food by ID"""
        try:
            result = self.client.table('foods').select('*').eq('id', food_id).execute()
            return result.data[0] if result.data else None
        except Exception as e:
            print(f"Error getting food: {e}")
            return None
    
    async def search_foods(self, query: str) -> List[Dict[str, Any]]:
        """Search foods by name"""
        try:
            result = self.client.table('foods').select('*').ilike('name', f'%{query}%').execute()
            return result.data or []
        except Exception as e:
            print(f"Error searching foods: {e}")
            return []

# Global instance
supabase_service = SupabaseService()

File: frontend/app/services/supabaseClient.ts
import { createClient } from "@supabase/supabase-js";

const supabaseUrl = 'https://dmskwpquomqcsumaqaej.supabase.co';
const supabaseAnonKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImRtc2t3cHF1b21xY3N1bWFxYWVqIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Mzg5Mzk0NDMsImV4cCI6MjA1NDUxNTQ0M30.A6aMcEqnl2V7l4vlwG9dUtK7F4BBIT3K4Lz7L-rHTd4';

export const supabase = createClient(supabaseUrl, supabaseAnonKey);

// Database service interfaces
export interface User {
  id: string;
  name: string;
  email: string;
  profile_image?: string;
  firebase_uid: string;
  created_at?: string;
  updated_at?: string;
}

export interface Restaurant {
  id: string;
  name: string;
  description: string;
  category: string;
  address: string;
  coordinates: {
    latitude: number;
    longitude: number;
  };
  image_url?: string;
  rating?: number;
  price_range?: string;
  hours?: string;
  phone?: string;
  created_at?: string;
  updated_at?: string;
}

export interface Food {
  id: string;
  name: string;
  description: string;
  price: number;
  image_url?: string;
  restaurant_id: string;
  category?: string;
  ingredients?: string[];
  created_at?: string;
  updated_at?: string;
}

// User service methods
export const userService = {
  async createUser(userData: Omit<User, 'id' | 'created_at' | 'updated_at'>): Promise<User | null> {
    const { data, error } = await supabase
      .from('users')
      .insert(userData)
      .select()
      .single();
    
    if (error) {
      console.error('Error creating user:', error);
      return null;
    }
    return data;
  },

  async getUserByFirebaseUid(firebaseUid: string): Promise<User | null> {
    const { data, error } = await supabase
      .from('users')
      .select('*')
      .eq('firebase_uid', firebaseUid)
      .single();
    
    if (error) {
      console.error('Error getting user:', error);
      return null;
    }
    return data;
  },

  async updateUser(id: string, updates: Partial<User>): Promise<User | null> {
    const { data, error } = await supabase
      .from('users')
      .update(updates)
      .eq('id', id)
      .select()
      .single();
    
    if (error) {
      console.error('Error updating user:', error);
      return null;
    }
    return data;
  }
};

// Restaurant service methods
export const restaurantService = {
  async getAllRestaurants(): Promise<Restaurant[]> {
    const { data, error } = await supabase
      .from('restaurants')
      .select('*')
      .order('name');
    
    if (error) {
      console.error('Error getting restaurants:', error);
      return [];
    }
    return data || [];
  },

  async getRestaurantById(id: string): Promise<Restaurant | null> {
    const { data, error } = await supabase
      .from('restaurants')
      .select('*')
      .eq('id', id)
      .single();
    
    if (error) {
      console.error('Error getting restaurant:', error);
      return null;
    }
    return data;
  },

  async searchRestaurants(query: string): Promise<Restaurant[]> {
    const { data, error } = await supabase
      .from('restaurants')
      .select('*')
      .ilike('name', `%${query}%`)
      .order('name');
    
    if (error) {
      console.error('Error searching restaurants:', error);
      return [];
    }
    return data || [];
  },

  async getRestaurantsByCategory(category: string): Promise<Restaurant[]> {
    const { data, error } = await supabase
      .from('restaurants')
      .select('*')
      .eq('category', category)
      .order('name');
    
    if (error) {
      console.error('Error getting restaurants by category:', error);
      return [];
    }
    return data || [];
  }
};

// Food service methods
export const foodService = {
  async getFoodsByRestaurant(restaurantId: string): Promise<Food[]> {
    const { data, error } = await supabase
      .from('foods')
      .select('*')
      .eq('restaurant_id', restaurantId)
      .order('name');
    
    if (error) {
      console.error('Error getting foods:', error);
      return [];
    }
    return data || [];
  },

  async getFoodById(id: string): Promise<Food | null> {
    const { data, error } = await supabase
      .from('foods')
      .select('*')
      .eq('id', id)
      .single();
    
    if (error) {
      console.error('Error getting food:', error);
      return null;
    }
    return data;
  },

  async searchFoods(query: string): Promise<Food[]> {
    const { data, error } = await supabase
      .from('foods')
      .select('*')
      .ilike('name', `%${query}%`)
      .order('name');
    
    if (error) {
      console.error('Error searching foods:', error);
      return [];
    }
    return data || [];
  }
};